index.html

```diff
@@ -274,6 +274,229 @@ <h3>
         </ol>
       </section>
     </section>
+    <section>
+      <h2>
+        `protocol_handlers` member
+      </h2>
+      <p>
+        The [=manifest's=] <code><dfn data-dfn-for=
+        "manifest">protocol_handlers</dfn></code> member is an array of
+        <a>protocol handler description</a>s that allows a web application to
+        handle URL protocols.
+      </p>
+      <p>
+        On installation, a user agent SHOULD register protocol handlers with
+        the Operating System via interactions that are consistent with:
+      </p>
+      <ul>
+        <li>If there are multiple registered handlers for a protocol, the OS
+        allows the user to select which app should open it, and also allows the
+        user to set a default app.
+        </li>
+        <li>clicking on a registered protocol will launch the registered
+        application. If this is the web app, then execute the [=invoke a
+        protocol handler=] steps defined in [=HTML=], where the user agent
+        SHOULD navigate to [=url=] and the appropriate browsing context is set
+        to a new top level browsing context.
+        </li>
+      </ul>
+      <aside class="note">
+        Protocol handlers could, for instance, be used for web app
+        communication where one app directly invokes another and passes data
+        via custom protocol links.
+      </aside>
+      <section>
+        <h3>
+          Processing the `protocol_handlers` member
+        </h3>
+        <p>
+          To <dfn>process the `protocol_handlers` member</dfn>, given
+          [=object=] |json:JSON|, |manifest:ordered map|, and |start URL:URL|:
+        </p>
+        <ol>
+          <li>Let <var>processedProtocolHandlers</var> be a new [=list=] of
+          |json:JSON|["<var>protocol_handlers</var>"].
+          </li>
+          <li>Set manifest["<var>protocol_handlers</var>"] to
+          <var>processedProtocolHandlers</var>.
+          </li>
+          <li>[=list/For each=] <var>protocol_handler</var> (<a>protocol
+          handler description</a>):
+            <ol>
+              <li>If |protocol_handler|["protocol"] or
+              <var>protocol_handler</var>["url"] is undefined,
+              [=iteration/continue=].
+              </li>
+              <li>Let (<dfn>normalizedProtocol</dfn>, <dfn>normalizedUrl</dfn>)
+              be the result of running [=normalize protocol handler
+              parameters=] with <var>protocol_handler</var>["protocol"], <var>
+                protocol_handler</var>["url"] using <var>manifest URL</var> as
+                the base URL, and [=this=]'s relevant [=environment settings
+                object=]. If the result is failure, [=iteration/continue=].
+              </li>
+              <li>If [=normalizedUrl=] is not [=manifest/within scope=] of
+              <var>start URL</var>, [=iteration/continue=].
+              </li>
+              <li>If [=normalizedUrl=] already exists in
```

Comment:
exists -> use [list/contains] instead? You'll have to reformat the sentence here I think (if the map contains the url, continue)